src/js/media-container.js

```diff
@@ -294,6 +307,29 @@ class MediaContainer extends window.HTMLElement {
   }
 }
 
+const mediaElPropTypes = {
```

Comment:
That's probably more prudent. I can get behind checking for play/pause.

> Somewhere there's a line where we're sending a "this media does not support X feature" warning/error not a "this media is not compatible".

I also like this approach of not failing silently, but logging the missing APIs.

Can we do the following:

- Check for play/pause, have that be required
- Log the other missing APIs in a warning.

Reply:
Sounds good. Though unless it's simple and obvious, we should probably boot the second bullet to another issue or to-do, since this PR is part of solving a specific issue.

Reply:
Just syncing with @cjpillsbury on this. I think this is a big more thorny.

We actually have a lot of code that relies on these properties for the internal state management (checking for `textTracks`, for example).

So essentially, this is not really valid use case. We've been pretty explicit from the beginning of "any media element must conform to the HTML Video element spec"

> Are there cases were you might want to be able to build a simple media element that just plays and pauses and just use the play button with it?

So now after talking it through, this "simple media element" isn't a valid use case unless we change our position that you don't need to use an element that conforms to the HTML Video element spec.

Reply:
Where's the line then? Does every media element have to implement the `.videoTracks` API and create their own version of media source extensions?

I don't know that just play/pause support is the right minimum, but requiring the entire video element API is unreasonable. We do want people to actually build media elements.

I think I'm more excited about making media container smart enough to handle when there's no textTracks property than making the bar to create a media element super high.

Reply:
I did try to call this out as a bigger topic. :) We don't protect against "unfinished" medias today, so this doesn't need to be a blocker to this PR, unless we want to make it one.

Reply:
That's definitely fair. I guess I saw the async as "sidestepping" the issue and the interface check as an acceptable "solving" of the issue (at least disallowing the scenario to happen as a rule) based on prior conversations, but I'll go ahead and remove the structural type checking predication for now so it's not a blocker.

Reply:
That makes sense to me. Should we move it to a discussion?

Reply:
I think it can be an issue at this point, since we've discussed it and there's some action items

https://github.com/muxinc/media-chrome/issues/94